packages/ui/src/proposals/components/ProposalList/ProposalList.tsx

```diff
@@ -1,28 +1,42 @@
 import React from 'react'
+import styled from 'styled-components'
 
 import { List } from '@/common/components/List'
 import { ListHeader, ListHeaders } from '@/common/components/List/ListHeader'
+import { Colors } from '@/common/constants'
 import { ProposalColLayout } from '@/proposals/constants'
-import { Proposal } from '@/proposals/types'
+import { PastProposal, Proposal } from '@/proposals/types'
 
 import { ProposalListItem } from './ProposalListItem'
 
-export interface ProposalListProps {
+export type ProposalListProps = CurrentProposalList | PastProposalList
+interface CurrentProposalList {
   proposals: Proposal[]
+  past?: false
+}
+interface PastProposalList {
+  proposals: PastProposal[]
+  past: true
```

Comment:
I think that you can use `Proposal` type with optional `endedAt`. This way the list will get a bit simpler.

Having only `isPast` type to decide which field to use to display a date:

```tsx
const pastProposals = usePastProposals()

<ProposalsList proposals={pastProposals} isPast />

// or

const proposal = useProposals()

<ProposalsList proposals={proposals} />
```